### ISSUE TITLE
PortraitUpsideDown Orientation not working

### ISSUE THREAD

**Issue by derekmulcahy** [futuretap/InAppSettingsKit#307]
I would like my application to run in PortraitUpsideDown mode in either iOS7 or iOS8. Currently the InAppSettingsKit view is always being displayed in Portrait mode regardless of the orientation of the device. This means that when the device is held with the Home button at the top then the settings page is upside down.

**Comment by futuretap** [comment]
IASK does not specify any orientation so it uses the system default `supportedInterfaceOrientations` which is `UIInterfaceOrientationMaskAll` for iPad and `UIInterfaceOrientationMaskAllButUpsideDown` for iPhone.

You should subclass `IASKAppSettingsViewController` and override `-supportedInterfaceOrientations` to return `UIInterfaceOrientationMaskAll`.

**Comment by derekmulcahy** [comment]
I added the supportedInterfaceOrientations to return UIInterfaceOrientationMaskAll as suggested. The method is called when the settings screen appears BUT the screen is still upside down.

I am testing an iPhone app in 2X mode on a iPad Air 2. Autorotate works as expected with or without the override in the iOS Simulator.

**Comment by futuretap** [comment]
You might want to test this with just an arbitrary `UIViewController`. I'm pretty sure IASK does nothing to prevent upside down mode.

**Comment by derekmulcahy** [comment]
I have found a solution to my problem by testing with a simplified program. The IASKAppSettingsViewController was added as the RootViewController for a UINavigationController. In this circumstance the following code must be added to a subclassed UINavigationController. The existing IASKAppSettingsViewController then just works.

Thanks for helping to track it down.
Derek

``` Objective-C
-(BOOL)shouldAutorotate
{
    return YES;
}

-(NSUInteger)supportedInterfaceOrientations
{
    return UIInterfaceOrientationMaskPortrait | UIInterfaceOrientationMaskPortraitUpsideDown;
}
```
